### DISCUSSION TITLE
Pre render does not work Vike + NX

### DISCUSSION THREAD

**Discussion by peeyushsingla** [vikejs/vike#1474] [Discussion] [answered]
HI Everyone,

We have one vike(vite-ssr-plugin) app as standalone app and now migrating it to our monorepo that using nx

We are having issue making pre-render work(html files stopped to generate after moving inside nx), 

To reproduce we tried creating two small app one using `vite + vike` and one with `nx + vite + vike` the exact same code without NX works perfectly but with NX it does not

We checked couple of issues on Github where we found out vike do support nx 

Could someone please help us:

* [Vite + vike app](https://github.com/peeyushsingla/vite-vike-ssr)
* [nx + vite + vike app](https://github.com/peeyushsingla/nx-vite-vike)
![Screenshot 2024-02-05 at 15 17 55](https://github.com/vikejs/vike/assets/5054529/30e7b942-7c99-461a-9895-dd9a6e018d9b)
![Screenshot 2024-02-05 at 15 18 31](https://github.com/vikejs/vike/assets/5054529/aba42be2-bd0c-4c8f-bf70-de0e7a0047dc)

**Comment by brillout** [comment]
How about manually triggering `$ vike prerender` yourself (see https://vike.dev/prerender-config#disableautorun)? Also, can you create an issue on Nx's GitHub? It's a Nx issue.

**Reply by peeyushsingla** [reply, marked as answer]
We replaced Nx vite build with the vite build directly and it fixed the issue


```
{
  "name": "my-app",
  "$schema": "../node_modules/nx/schemas/project-schema.json",
  "sourceRoot": "my-app/src",
  "projectType": "application",
  "targets": {
    "build": {
      "executor": "nx:run-commands",
      "outputs": [
        "{options.outputPath}"
      ],
      "defaultConfiguration": "production",
      "options": {
        "cwd": "my-app",
        "command": "vite build",
        "outputPath": "dist/my-app"
      },
      "configurations": {
        "development": {
          "mode": "development"
        },
        "production": {
          "mode": "production"
        }
      }
    },
    "serve": {
      "executor": "@nx/vite:dev-server",
      "defaultConfiguration": "development",
      "options": {
        "buildTarget": "my-app:build"
      },
      "configurations": {
        "development": {
          "buildTarget": "my-app:build:development",
          "hmr": true
        },
        "production": {
          "buildTarget": "my-app:build:production",
          "hmr": false
        }
      }
    },
    "preview": {
      "executor": "@nx/vite:preview-server",
      "defaultConfiguration": "development",
      "options": {
        "buildTarget": "my-app:build"
      },
      "configurations": {
        "development": {
          "buildTarget": "my-app:build:development"
        },
        "production": {
          "buildTarget": "my-app:build:production"
        }
      }
    },
    "test": {
      "executor": "@nx/vite:test",
      "outputs": [
        "{options.reportsDirectory}"
      ],
      "options": {
        "reportsDirectory": "../coverage/my-app"
      }
    },
    "lint": {
      "executor": "@nx/eslint:lint"
    }
  },
  "tags": []
}

```
